src/fidesops/api/v1/endpoints/drp_endpoints.py

```diff
@@ -0,0 +1,154 @@
+import json
+import logging
+from typing import Dict, Any, Optional
+
+import jwt
+from fastapi import HTTPException, Depends, APIRouter, Security
+from sqlalchemy.orm import Session
+from starlette.status import (
+    HTTP_404_NOT_FOUND,
+    HTTP_422_UNPROCESSABLE_ENTITY,
+    HTTP_424_FAILED_DEPENDENCY,
+    HTTP_200_OK,
+    HTTP_400_BAD_REQUEST,
+)
+
+from fidesops import common_exceptions
+from fidesops.api import deps
+from fidesops.api.v1 import scope_registry as scopes
+from fidesops.api.v1 import urn_registry as urls
+from fidesops.core.config import config
+from fidesops.models.policy import Policy
+from fidesops.models.privacy_request import PrivacyRequest
+from fidesops.schemas.drp_privacy_request import DrpPrivacyRequestCreate, DrpIdentity
+from fidesops.schemas.privacy_request import PrivacyRequestDRPStatusResponse
+from fidesops.schemas.redis_cache import PrivacyRequestIdentity
+from fidesops.service.drp.drp_fidesops_mapper import DrpFidesopsMapper
+from fidesops.service.privacy_request.request_runner_service import PrivacyRequestRunner
+from fidesops.service.privacy_request.request_service import (
+    build_required_privacy_request_kwargs,
+    cache_data,
+)
+from fidesops.util.cache import FidesopsRedis
+from fidesops.util.oauth_util import verify_oauth_client
+
+logger = logging.getLogger(__name__)
+router = APIRouter(tags=["DRP"], prefix=urls.V1_URL_PREFIX)
+EMBEDDED_EXECUTION_LOG_LIMIT = 50
+
+
+@router.post(
+    urls.DRP_EXERCISE,
+    status_code=HTTP_200_OK,
+    response_model=PrivacyRequestDRPStatusResponse,
+)
+def create_drp_privacy_request(
+    *,
+    cache: FidesopsRedis = Depends(deps.get_cache),
+    db: Session = Depends(deps.get_db),
+    data: DrpPrivacyRequestCreate,
+) -> PrivacyRequestDRPStatusResponse:
+    """
+    Given a drp privacy request body, create and execute
+    a corresponding Fidesops PrivacyRequest
+    """
+
+    jwt_key: str = config.security.DRP_JWT_SECRET
+    if jwt_key is None:
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail="JWT key must be provided",
```

Comment:
This isn't really a `400` since if this variable isn't in the config it represents a server misconfiguration. `501_NOT_IMPLEMENTED` or `500_SERVER_ERROR` might be best.

Reply:
ah right, good point